src/services/formatting/formatting.ts

```diff
@@ -394,6 +394,11 @@ namespace ts.formatting {
                 const startLinePosition = getLineStartPositionForPosition(startPos, sourceFile);
                 const column = SmartIndenter.findFirstNonWhitespaceColumn(startLinePosition, startPos, sourceFile, options);
                 if (startLine !== parentStartLine || startPos === column) {
+                    // Use the base indent size if it is greater than
+                    // the indentation of the inherited predecessor.
+                    if (options.BaseIndentSize > column) {

```

Comment:
undefined or null will return false in this case as well so didn't think it was necessary.  But perhaps better for consistency - I'll export the function and use it here as well.